src/CombatManager.cpp

```diff
@@ -83,7 +83,49 @@ void CombatManager::pcAttackNpcs(CNSocket *sock, CNPacketData *data) {
     }
 }
 
-void CombatManager::combatBegin(CNSocket *sock, CNPacketData *data) {} // stub
+void CombatManager::combatBegin(CNSocket *sock, CNPacketData *data) {
+    sP_CL2FE_REQ_PC_COMBAT_BEGIN* pkt = (sP_CL2FE_REQ_PC_COMBAT_BEGIN*)data->buf;
+    Player* plr = PlayerManager::getPlayer(sock);
+
+    int32_t* pktdata = (int32_t*)((uint8_t*)data->buf + sizeof(sP_CL2FE_REQ_PC_COMBAT_BEGIN));
```

Comment:
`sP_CL2FE_REQ_PC_COMBAT_BEGIN` is not a variable-length packet. This `pktdata` variable will always point to garbage.

In fact, I think defining `pkt` might not even be necessary here, since it only contains the `iPC_ID`, and `REQ` packets are not typically sent to other players, so it's probably redundant. I could be wrong, though.